fcrepo-kernel-impl/src/test/java/org/fcrepo/kernel/impl/services/DeleteResourceServiceImplTest.java

```diff
@@ -0,0 +1,192 @@
+    }
+
+    @Test
+    public void testContainerDelete() throws Exception {
+        when(container.getId()).thenReturn(RESOURCE_ID);
+        when(container.isAcl()).thenReturn(false);
+        when(container.getAcl()).thenReturn(null);
+
+        when(containmentIndex.getContainedBy(eq(tx), eq(container))).thenReturn(Stream.<String>builder().build());
+
+        final ArgumentCaptor<DeleteResourceOperation> containerOperationCaptor = ArgumentCaptor.forClass(DeleteResourceOperation.class);
+        service.perform(tx, container);
+        verifyResourceOperation(RESOURCE_ID, containerOperationCaptor, pSession);
+        verify(containmentIndex).getContainedBy(eq(tx), eq(container));
+    }
+
+    @Test
+    public void testRecursiveDelete() throws Exception {
+        when(container.getId()).thenReturn(RESOURCE_ID);
+        when(container.getId()).thenReturn(RESOURCE_ID);
+        when(container.isAcl()).thenReturn(false);
+        when(container.getAcl()).thenReturn(null);
+        when(childContainer.getId()).thenReturn(CHILD_RESOURCE_ID);
+        when(childContainer.isAcl()).thenReturn(false);
+        when(childContainer.getAcl()).thenReturn(null);
+
+        when(resourceFactory.getResource(tx, CHILD_RESOURCE_ID)).thenReturn(childContainer);
+        when(containmentIndex.getContainedBy(eq(tx), eq(container))).thenReturn(Stream.<String>builder()
+                                                                                      .add(CHILD_RESOURCE_ID).build());
+        when(container.isAcl()).thenReturn(false);
+        when(container.getAcl()).thenReturn(null);
+        final ArgumentCaptor<DeleteResourceOperation> captor = ArgumentCaptor.forClass(DeleteResourceOperation.class);
+        service.perform(tx, container);
+
+        verify(pSession, times(2)).persist(captor.capture());
+        final List<DeleteResourceOperation> operations = captor.getAllValues();
+        assertEquals(2, operations.size());
+
+        assertEquals(CHILD_RESOURCE_ID, operations.get(0).getResourceId());
+        assertEquals(RESOURCE_ID, operations.get(1).getResourceId());
+
+        verify(containmentIndex).getContainedBy(eq(tx), eq(container));
+    }
+
+    private void verifyResourceOperation(final String resourceId,
+                                         final ArgumentCaptor<DeleteResourceOperation> captor,
+                                         final PersistentStorageSession pSession) throws Exception {
+        verify(pSession).persist(captor.capture());
+        final DeleteResourceOperation containerOperation = captor.getValue();
+        assertEquals(resourceId, containerOperation.getResourceId());
+    }
+
+    @Test
+    public void testAclDelete() throws Exception {
+        when(acl.getId()).thenReturn(RESOURCE_ACL_ID);
+        when(acl.isAcl()).thenReturn(true);
+        final ArgumentCaptor<DeleteResourceOperation> aclCaptor = ArgumentCaptor.forClass(DeleteResourceOperation.class);
```

Comment:
I can't remember if these were here before, but I'd suggest using the `@Captor` annotation since this is in all the tests. See:
https://www.baeldung.com/mockito-annotations#captor-annotation

Reply:
done.